scripts/main.py

```diff
@@ -0,0 +1,150 @@
+import pandas as pd
+from sklearn.cluster import DBSCAN
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+# NLTK variables to aid in cleaning/preprocessing
+ps = nltk.PorterStemmer()
+
+# Pandas dataframe for tweets
+df = pd.DataFrame(columns=["id", "tweet", "popularity"])
+
+
+class Results:
+    def __init__(self):
+        """Initialize final results of the analysis"""
+        self.clusters_count = df.clusters.value_counts()
+        self.df_results = df.groupby(["clusters"]).max().reset_index()
+        print("Number of tweets per cluster: \n{}".format(self.clusters_count))
+        print("Top cluster tweets: \n{}".format(self.df_results.to_string()))
+
+
+def authorise_api():
+    """Authorise access to twitter API and return the api handler"""
+    load_dotenv()
+    consumer_key = os.getenv("CONSUMER_KEY")
+    consumer_key_secret = os.getenv("CONSUMER_KEY_SECRET")
+    access_token = os.getenv("ACCESS_TOKEN")
+    access_token_secret = os.getenv("ACCESS_TOKEN_SECRET")
+
+    auth = tw.OAuthHandler(consumer_key, consumer_key_secret)
+    auth.set_access_token(access_token, access_token_secret)
+    api = tw.API(auth, wait_on_rate_limit=True)
+
+    return api
+
+
+def get_top_trends(api):
+    """Returns a list of top trends at the specified location"""
+    # Current location: New Delhi, India. WOEID = 1 for global trending
+    latitude = 28.644800
+    longitude = 77.216721
+
+    locations = api.trends_closest(latitude, longitude)
+    woeid = locations[0]["woeid"]
+
+    trends = api.trends_place(woeid)
+    trends_dict = trends[0]["trends"]
+
+    return [trends_dict[0]]
+
+
+def de_emojify(text):
+    """Remove emoticons from given text"""
+    regrex_pattern = re.compile(pattern="["
+                                        u"\U0001F600-\U0001F64F"  # emoticons
+                                        u"\U0001F300-\U0001F5FF"  # symbols & pictographs
+                                        u"\U0001F680-\U0001F6FF"  # transport & map symbols
+                                        u"\U0001F1E0-\U0001F1FF"  # flags (iOS)
+                                        u"\U00002500-\U00002BEF"  # chinese char
+                                        u"\U00002702-\U000027B0"
+                                        u"\U00002702-\U000027B0"
+                                        u"\U000024C2-\U0001F251"
+                                        u"\U0001f926-\U0001f937"
+                                        u"\U00010000-\U0010ffff"
+                                        u"\u2640-\u2642"
+                                        u"\u2600-\u2B55"
+                                        u"\u200d"
+                                        u"\u23cf"
+                                        u"\u23e9"
+                                        u"\u231a"
+                                        u"\ufe0f"  # dingbats
+                                        u"\u3030"
+                                        "]+", flags=re.UNICODE)
```

Comment:
Isn't there any alternative to do this? Please check if there exists any alternative

Reply:
Not really. This was the best solution to removing emojis. All other more concise methods do not end up encapsulating all emojis.